doc/developer_guide/aws.md

```diff
@@ -0,0 +1,160 @@
+* clones the script-languages-release repository
+* changes the netplan configuration. This is necessary to have proper network configuration when running the VM image
+
+Finally, the default password will be set, and also the password will be marked as expired, such that the user will be forced to enter a new password during initial login.
+Also, the ssh password authentication will be enabled, and for security reasons the folder "~/.ssh" will be removed.
+
+### Tests
+
+DSS comes with a number of tests in directory `test`.
+There are subdirectories clustering tests with common scope and prerequisites e.g. external resources.
+
+| Directory           | Content |
+|---------------------|---------|
+| `test/unit`         | Simple unit tests requiring no additional setup or external resources. |
+| `test/integration`  | Integration tests with longer runtime and some requiring additional resources. |
+| `test/aws`          | Tests involving AWS resources.  In order to execute these tests you need an AWS account, a user with permissions in this account, and an access key. |
+
+To run the tests in file `test/integration/test_ci.py` please use
+```shell
+export DSS_RUN_CI_TEST=true
+poetry run test/integration/test_ci.py
+```
+
+Creating a docker image is quite time-consuming, currently around 7 minutes. In order to use an existing docker image in the tests in `integration/test_create_dss_docker_image.py` simply add CLI option `--dss-docker-image` when calling `pytest`:
+
+```shell
+poetry run pytest --dss-docker-image exasol/data-science-sandbox:0.1.0
+```
+
+#### Executing tests involving AWS resources
+
+In AWS web interface, IAM create an access key for CLI usage and save or download the *access key id* and the *secret access key*.
+
+In file `~/.aws/config` add lines
+```
+[profile dss_aws_tests]
+region = eu-central-1
+```
+
+In file `~/.aws/credentials` add
+```
+[dss_aws_tests]
+aws_access_key_id=...
+aws_secret_access_key=...
+```
+
+In case your are using MFA authentication please allocate a temporary token.
+
+After that you can set an environment variable and execute the tests involving AWS resources:
+
+```shell
+export AWS_PROFILE=dss_aws_tests_mfa
+poetry run pytest test/test_deploy_codebuild.py
+```
+
+#### Executing tests involving Ansible
+
+For making pytest display Ansible log messages, please use
+
+```shell
+poetry run pytest -s -o log_cli=true -o log_cli_level=INFO
+```
+
+### Ansible packages
+
+The packages to be installed by Ansible are using pinned versions, e.g. for [poetry](../../exasol/ds/sandbox/runtime/ansible/roles/poetry/defaults/main.yml).
+
+In case ansible reports "no available installation candidate" for a specific version of a package, please search for newer versions of the package on https://packages.ubuntu.com/.
+
+If the update is very new and not yet displayed on packages.ubuntu.com you can use
+
+```shell
+apt-get update
+apt-search <package>
+```
+
```

Comment:
Sections "Executing tests involving Ansible" and "Ansible packages" should also go to general developer guide.